### PR TITLE
Allow customer_id to be specified in identify URL

### PR DESCRIPTION
**Pull request by jrbeck**
This is a rebase of the work here:

https://github.com/trwalzer/customerio-ruby

The issue is here:

https://github.com/customerio/customerio-ruby/issues/78

I did nothing new, but I wanted to use the technique @trwazler added, but with a more recent version of the gem.

### CODE REVIEW

**Comment by jeremyw** [comment]
Can this be merged? We're running into this issue as well -- the API does what we need but the Ruby wrapper doesn't.

File: spec/client_spec.rb [hunk 1]
```diff
@@ -183,6 +183,14 @@ def json(data)
 
       lambda { client.identify(attributes) }.should_not raise_error()
     end
+
+    it "uses provided url_id rather than id" do
```
Comment (jeremyw) [inline spec/client_spec.rb:187]:
```suggestion
    it "uses provided customer_id rather than id" do
```

**Comment by jrbeck** [comment]
@jeremyw this is ready to be merged, just waiting for the review. @stephen-pope-customer-io can we get this reviewed?

**Comment by stephen-pope-customer-io** [comment]
You surely mean @hownowstephen , and not me here

**Comment by jrbeck** [comment]
@stephen-pope-customer-io, thank you!

**Comment by richdawe-cio** [comment]
@jrbeck @jeremyw Thanks for your changes and the PR.

I implemented a slightly different approach to the one in this PR, which was to make the `identify` automatically use the `cio_id` attributed as the customer ID. You can see the PR that I merged, and an example usage at https://github.com/customerio/customerio-ruby/pull/109

I've published version 5.2.0 of the library in Ruby Gems. I'm happy to revisit it if it's not doing what you need. Could you take a look at this approach and the new library version, and let me know your thoughts, please?

**Comment by jeremyw** [comment]
@richdawe-cio I don't think that change solves this problem. 

We need to be able to pass email as the identifier in the path, and the id in the body.

**Comment by richdawe-cio** [comment]
> @richdawe-cio I don't think that change solves this problem.
> 
> We need to be able to pass email as the identifier in the path, and the id in the body.

Thanks for the quick feedback, and clarifying @jeremyw , I appreciate it . That wasn't clear from the original issue or the code changes in this PR. I'll add that feature in too.

**Comment by richdawe-cio** [comment]
Thanks for opening this PR. In the end, we implemented this in a slightly different way in https://github.com/customerio/customerio-ruby/pull/111 , but adding a `customer_id` parameter to the `identify` method. E.g.:

```ruby
$customerio.identify(
  :customer_id => "bob@example.com",
  :location => "Australia"
)
```

Thank you for opening this PR, the discussions on it, and the help resolving this issue with the Ruby client library.

I don't think this PR is needed any longer, so I'm going to close it.

**Pull request closed without merging**